internal/v5/resources.go

```diff
@@ -40,6 +46,42 @@ func (h *ReqHandler) serveResources(id *router.ResolvedURL, w http.ResponseWrite
 	}
 }
 
+func (h *ReqHandler) serveDockerResourceUploadInfo(id *router.ResolvedURL, w http.ResponseWriter, req *http.Request) error {
+	if err := h.AuthorizeEntityForOp(id, req, OpWrite); err != nil {
+		return errgo.Mask(err, errgo.Any)
+	}
+	resourceName := req.Form.Get("resource-name")
+	if resourceName == "" {
+		return badRequestf(nil, "must specify resource-name parameter")
+	}
+	e, err := h.Cache.Entity(&id.URL, charmstore.FieldSelector("charmmeta"))
+	if err != nil {
+		// Should never happen, as the entity will have been cached
+		// when the charm URL was resolved.
+		return errgo.Mask(err, errgo.Is(params.ErrNotFound))
+	}
+	if !charmstore.IsKubernetesCharm(e.CharmMeta) {
+		return errgo.WithCausef(nil, params.ErrForbidden, "%q does not support docker resource upload", id.URL.String())
+	}
+	r, ok := e.CharmMeta.Resources[resourceName]
+	if !ok {
+		return errgo.WithCausef(nil, params.ErrForbidden, "%q has no resource named %q", id.URL.String(), resourceName)
+	}
+	if r.Type != resource.TypeDocker {
```

Comment:
Just checking but you can upload a k8s charm without specifying first it's resources ?

Reply:
Yeah; that's true of any charm - you can't upload resources to a charm that doesn't yet exist.